experimental/hip/dynamic_symbols.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_EXPERIMENTAL_HIP_DYNAMIC_SYMBOLS_H_
+#define IREE_EXPERIMENTAL_HIP_DYNAMIC_SYMBOLS_H_
+
+#include "experimental/hip/hip_headers.h"
+#include "iree/base/api.h"
+#include "iree/base/internal/dynamic_library.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// iree_dynamic_library_t allows dynamically loading a subset of HIP driver API.
+// We load all the symbols in `dynamic_symbol_tables.h` and fail if any of the
+// symbol is not available. The functions signatures are matching the declarations
```

Comment:
Formatting seems off too here. We need to run clang-format over all these files..